src/java.base/share/classes/java/io/PrintStream.java

```diff
@@ -104,11 +105,22 @@ private static Charset toCharset(String csn)
         }
     }
 
+    /**
+     * Returns the charset used in this {@code PrintStream} instance.
+     *
+     * @since 18
+     * @return the charset used in this {@code PrintStream} instance
+     */
+    public Charset charset() {
+        return charset;
+    }
```

Comment:
This looks good. You could use {@return the charset used ...} to avoid repeating the message. Also might be better to move the method to after the constructors so that it's with the other instance methods.
The update method descriptions in PS, PW, and OutputStreamWriter look good.
So overall I think we've got to a good place. Wrapping a PS with PW and not inheriting the charset is an potential accident that goes back 20+ years.

Reply:
Thanks, Alan. Modified as suggested.